src/prelude/internals.mo

```diff
@@ -612,10 +612,10 @@ func @timer_helper() : async () {
   ignore (prim "global_timer_set" : Nat64 -> Nat64) exp;
   if (exp == 0) @timers := null;
 
-  for (o in thunks.vals()) {
+  label out for (o in thunks.vals()) {
     switch o {
-      case (?thunk) { ignore thunk() };
-      case _ { }
+      case (?thunk) ignore thunk();
+      case _ break out
```

Comment:
I think you could also just return  without the label and break

Reply:
I think the real overhead is the messaging cost (regardless of cycles)

Reply:
Right, but in the grand scheme of things this helps a bit. Why burn unnecessarily?